core/src/main/java/org/apache/iceberg/ReachableFileUtil.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.relocated.com.google.common.collect.Sets;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ReachableFileUtil {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ReachableFileUtil.class);
+
+  private ReachableFileUtil() {
+  }
+
+  /**
+   * Returns the location of the version.text file
+   *
+   * @param ops tableOperation for which version.text path needs to be retrieved
+   * @return the location of the version hint file
+   */
+  public static String versionHintLocation(TableOperations ops) {
+    return ops.metadataFileLocation("version-hint.text");
```

Comment:
Per @kbendick suggested, it is a good time to make a constant for "version-hint.text". But I'm OK to do this in another PR.